### PR TITLE
Draft: Implemented Calendar List View 

### PR DESCRIPTION
**Pull request by tylersocholotuik**
## What issue is this referencing?

#394 specifically, [this comment ](https://github.com/devedmonton/DES-Website/issues/394#issuecomment-2452052482)

Here is the first iteration of the idea @dgmouris had for the calendar list view. The events should be grouped by month, but there are unfortunately only events in November at the moment so I am not certain this works. 

Let me know your thoughts on anything that could be improved!

list-vew

## Do these code changes work locally and have you tested that they fix the issue yourself?

-   [x] yes!

## Does the following command run without warnings or errors?

-   [x] yes! `npm run pr-checks`

## Have you taken a look at our [contributing guidelines](https://github.com/devedmonton/DES-Website/blob/main/.github/CONTRIBUTING.md)?

-   [x] yes!

## My node version matches the one suggested when running `nvm use`?

-   [x] yes!

### CODE REVIEW

**Comment by tylersocholotuik** [comment]
Update: I fixed the height of the cards based on feedback from @johnjovero98, and I set the default view to list on mobile (<768px).

**Review by MandyMeindersma — APPROVED** [review]

**Comment by MandyMeindersma** [comment]
LOVE this! I am going to let another person look this over too. 

I am a BE dev by day so my FE is a little rusty

**Comment by arashsheyda** [comment]
the card width on mobile view is not quite responsive I think. also if we use `computed` for isMobile it would be better (I think we can even combine `selectedView` with `defaultView` using a [computed](https://vuejs.org/guide/essentials/computed.html#writable-computed))

Screenshot 2024-11-05 at 7 33 16 PM

**Comment by tylersocholotuik** [comment]
Ah ok I will try to fix that. I think the calendar view has the same issue on mobile as well.

**Comment by arashsheyda** [comment]
@tylersocholotuik thanks! (it's not a big deal though, if you ask me. we can merge it for now and add a fix later. looks great! thanks)

**Comment by tylersocholotuik** [comment]
> @tylersocholotuik thanks! (it's not a big deal though, if you ask me. we can merge it for now and add a fix later. looks great! thanks)

Ok sounds good! If I have some time later today I will try to tackle the responsive issue. Changing isMobile and selected view to computed caused an issue, however. 

`const isMobile = computed(() => useMediaQuery('(max-width: 768px)').value)`
`const selectedView = computed(() => isMobile.value ? 'list' : 'calendar')`

This will now change the view automatically based on the window size, and clicking on the tabs no longer works. Is there something I am missing? 

My original idea was that the default view is set only when the page loads based on the window size, then the user will have control with the tabs after that. If we want both (automatic switching + user control with tabs), I'm not sure how to get the tabs to work now.

**Comment by arashsheyda** [comment]
`selectedView` should stay `ref` in this case.

**Comment by tylersocholotuik** [comment]
> `selectedView` should stay `ref` in this case.

ok, so keep isMobile as computed, but change the others back to the way it was before?

`const isMobile = computed(() => useMediaQuery('(max-width: 768px)').value)`
`const defaultView = isMobile.value ? 'list' : 'calendar'`
`const selectedView = ref(defaultView)`

I could put the expression for defaultView in the ref if you prefer few lines.

**Comment by arashsheyda** [comment]
@tylersocholotuik my bad, I didn't notice `useMediaQuery` is from `@vueuse` and it is a computed already so no need for that. but how about something lik this:
```
const isMobile = useMediaQuery('(max-width: 768px)')
const calendarView = useCookie('calendarView', { default: () => 'calendar' })
const selectedView = computed({
  get: () => isMobile.value ? 'list' : calendarView.value,
  set: value => calendarView.value = value,
})
```

this way we save the users choice (in cookies) also do you think we should hide the tabs in mobile view since it wouldn't change the view anyway?
sorry again for the confusion about `useMediaQuery`

File: components/app/Calendar.vue
Comment (arashsheyda) [inline components/app/Calendar.vue:140]: can we use the same border colour as other places? thanks

```suggestion
            class="border-2 border-gray-400/40 rounded mb-4 break-word"
```

File: components/app/Calendar.vue
Comment (arashsheyda) [inline components/app/Calendar.vue:143]:
```suggestion
              class="p-4 flex justify-between items-center border-b rounded-t border-gray-400/40">
```

**Comment by tylersocholotuik** [comment]
@arashsheyda thanks for the help with the view selection, I have not learned about cookies yet. I made that change, plus hid the tabs on mobile view and changed the border colors.

**Comment by tylersocholotuik** [comment]
I realized I may have commited too hastily. I had the thought that maybe it isn't a good idea to prevent the option of using the calendar view on mobile? Or maybe I can change the media query to a smaller size where the calendar view really doesn't look great.

**Comment by dgmouris** [comment]
@arashsheyda thanks for the thorough review and @tylersocholotuik for the work. 

On mobile, the calendar view might need some work, and it's a bit difficult to use (I think it'll be some vue-cal config options).

I think the cookie is great but as long as we have the default as the list view that's awesome.

I'm a fan of this implementation
```
const isMobile = useMediaQuery('(max-width: 768px)')
const calendarView = useCookie('calendarView', { default: () => 'calendar' })
const selectedView = computed({
  get: () => isMobile.value ? 'list' : calendarView.value,
  set: value => calendarView.value = value,
})
```

File: components/app/Calendar.vue
Comment (dgmouris) [inline components/app/Calendar.vue:164]: @tylersocholotuik sometimes the events are going to have a bit of a longer description because of markdown from.

This technique I wouldn't use all the time but I think this will solve our problem and look good.

```suggestion
            <div
              class="p-4 event-description"
              :class="{ 'max-w-sm': isMobile }"
            >
```

The mobile view on pixel 7:
image

The desktop view:
image

@arashsheyda @tylersocholotuik let me know what you think.

I'd be ready to merge this once this implemented so that it doesn't break on mobile formats.

File: components/app/Calendar.vue
Comment (dgmouris) [inline components/app/Calendar.vue:41]: Awesome I like the use of using a computed value to group the events. I personally might change the `acc` on line 26 to something like `monthEvents` but not a show stopper whatsoever.

**Review by dgmouris — APPROVED** [review]
Once the above is fixed I'm good with merging this in:)

Great job @tylersocholotuik and thanks @arashsheyda, @MandyMeindersma for reviewing this too:)

**Comment by tylersocholotuik** [comment]
@dgmouris Thanks a lot for all the suggestions! I made all the changes you suggested and I also made the time and date on the cards move underneath the title header on mobile view to help with the width overflow issue. I also decided to do the media query at 700px rather than 768px because I think people would like to use the calendar view on tablets. Let me know if you agree with that.

Here is the new mobile view 
image

**Pull request merged**
